src/wp-includes/canonical.php

```diff
@@ -202,23 +235,37 @@ function redirect_canonical( $requested_url = null, $do_redirect = true ) {
 				$redirect['query'] = remove_query_arg( 'm', $redirect['query'] );
 			}
 			// now moving on to non ?m=X year/month/day links
-		} elseif ( is_day() && get_query_var( 'year' ) && get_query_var( 'monthnum' ) && ! empty( $_GET['day'] ) ) {
-			if ( $redirect_url = get_day_link( get_query_var( 'year' ), get_query_var( 'monthnum' ), get_query_var( 'day' ) ) ) {
-				$redirect['query'] = remove_query_arg( array( 'year', 'monthnum', 'day' ), $redirect['query'] );
-			}
-		} elseif ( is_month() && get_query_var( 'year' ) && ! empty( $_GET['monthnum'] ) ) {
-			if ( $redirect_url = get_month_link( get_query_var( 'year' ), get_query_var( 'monthnum' ) ) ) {
-				$redirect['query'] = remove_query_arg( array( 'year', 'monthnum' ), $redirect['query'] );
-			}
-		} elseif ( is_year() && ! empty( $_GET['year'] ) ) {
-			if ( $redirect_url = get_year_link( get_query_var( 'year' ) ) ) {
-				$redirect['query'] = remove_query_arg( 'year', $redirect['query'] );
+		} elseif ( is_date() ) {
+			$year  = get_query_var( 'year' );
+			$month = get_query_var( 'monthnum' );
+			$day   = get_query_var( 'day' );
+
+			if ( is_day() && get_query_var( 'year' ) && get_query_var( 'monthnum' ) && ! empty( $_GET['day'] ) ) {
```

Comment:
We can use the variables `$year` and `$month`. These are already declared.

Reply:
Agreed, this is the same upstream.